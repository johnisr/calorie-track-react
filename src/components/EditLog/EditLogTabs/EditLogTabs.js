import React from 'react';
import { connect } from 'react-redux';
import {Tabs, TabList, Tab, TabPanels, TabPanel} from '../../Tabs/Tabs';
import AddFoodFormToCurrentLog from './AddFoodFormToCurrentLog/AddFoodFormToCurrentLog';
import EditFoodFormForCurrentLog from './EditFoodFormForCurrentLog/EditFoodFormForCurrentLog';
import CurrentLogAddFoodListContainer from './CurrentLogAddFoodListContainer/CurrentLogAddFoodListContainer';
import FoodListFilters from '../../FoodListFilters/FoodListFilters';

export const EditLogTabs = (props) => (
  <Tabs activeIndex={props.activeIndex}>
    <TabList>
      <Tab>Add from database</Tab>
      <Tab>Add Manually</Tab>
      <Tab>Edit</Tab>
    </TabList>

    <TabPanels>
      <TabPanel>
        <FoodListFilters />
        <CurrentLogAddFoodListContainer />
      </TabPanel>
      <TabPanel>
        <AddFoodFormToCurrentLog />
      </TabPanel>
      <TabPanel>
        <EditFoodFormForCurrentLog />
      </TabPanel>
    </TabPanels>
  </Tabs>
);

const mapStateToProps = (state, props) => ({
  activeIndex: state.currentLog.editFoodIndex !== undefined ? 2 : 0,
});

export default connect(mapStateToProps)(EditLogTabs);