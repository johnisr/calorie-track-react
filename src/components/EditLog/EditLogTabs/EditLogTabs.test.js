import React from 'react';
import { shallow } from 'enzyme';
import { EditLogTabs } from './EditLogTabs';

it('should render correctly', () => {
  const wrapper = shallow(<EditLogTabs activeIndex={2} />);
  expect(wrapper).toMatchSnapshot();
});