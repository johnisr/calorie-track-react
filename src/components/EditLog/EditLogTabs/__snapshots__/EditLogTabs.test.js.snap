// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    activeIndex={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeIndex": 2,
      "children": Array [
        <Unknown>
          <Unknown>
                    Add from database
          </Unknown>
          <Unknown>
                    Add Manually
          </Unknown>
          <Unknown>
                    Edit
          </Unknown>
</Unknown>,
        <Unknown>
          <Unknown>
                    <Connect(FoodListFilters) />
                    <Connect(CurrentLogAddFoodListContainer) />
          </Unknown>
          <Unknown>
                    <Connect(AddFoodFormToCurrentLog) />
          </Unknown>
          <Unknown>
                    <Connect(EditFoodFormForCurrentLog) />
          </Unknown>
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              Add from database
</Unknown>,
            <Unknown>
              Add Manually
</Unknown>,
            <Unknown>
              Edit
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Add from database",
            },
            "ref": null,
            "rendered": "Add from database",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Add Manually",
            },
            "ref": null,
            "rendered": "Add Manually",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Edit",
            },
            "ref": null,
            "rendered": "Edit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              <Connect(FoodListFilters) />
              <Connect(CurrentLogAddFoodListContainer) />
</Unknown>,
            <Unknown>
              <Connect(AddFoodFormToCurrentLog) />
</Unknown>,
            <Unknown>
              <Connect(EditFoodFormForCurrentLog) />
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Connect(FoodListFilters) />,
                <Connect(CurrentLogAddFoodListContainer) />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(AddFoodFormToCurrentLog) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(EditFoodFormForCurrentLog) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeIndex": 2,
        "children": Array [
          <Unknown>
            <Unknown>
                        Add from database
            </Unknown>
            <Unknown>
                        Add Manually
            </Unknown>
            <Unknown>
                        Edit
            </Unknown>
</Unknown>,
          <Unknown>
            <Unknown>
                        <Connect(FoodListFilters) />
                        <Connect(CurrentLogAddFoodListContainer) />
            </Unknown>
            <Unknown>
                        <Connect(AddFoodFormToCurrentLog) />
            </Unknown>
            <Unknown>
                        <Connect(EditFoodFormForCurrentLog) />
            </Unknown>
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                Add from database
</Unknown>,
              <Unknown>
                Add Manually
</Unknown>,
              <Unknown>
                Edit
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Add from database",
              },
              "ref": null,
              "rendered": "Add from database",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Add Manually",
              },
              "ref": null,
              "rendered": "Add Manually",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Edit",
              },
              "ref": null,
              "rendered": "Edit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                <Connect(FoodListFilters) />
                <Connect(CurrentLogAddFoodListContainer) />
</Unknown>,
              <Unknown>
                <Connect(AddFoodFormToCurrentLog) />
</Unknown>,
              <Unknown>
                <Connect(EditFoodFormForCurrentLog) />
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Connect(FoodListFilters) />,
                  <Connect(CurrentLogAddFoodListContainer) />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(AddFoodFormToCurrentLog) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(EditFoodFormForCurrentLog) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
